### PR TITLE
Different mime-types for batch uploads

### PR DESCRIPTION
**Pull request by tillgeissler**
This PR refactors the implementation for enabling batch upload with different mime types per document as well as the automatic guessing of the type. 
There is also a minor change to only upload the file name instead of the whole path to not expose the users directory structure.

### CODE REVIEW

File: sap_business_document_processing/document_information_extraction_client/dox_api_client.py [hunk 1]
```diff
@@ -178,7 +180,8 @@ def extract_information_from_document(self, document_path: str, client_id, docum
         :param document_path: The path to the document
         :param client_id: The client ID for which the document should be uploaded
         :param document_type: The type of the document being uploaded. For available document types see documentation
-        :param mime_type: Content type of the uploaded file. If None is given, the content type is fetched automatically.
+        :param mime_type: Content type of the uploaded file. If 'unknown' is given, the content type is fetched
+        automatically. Default is 'application/pdf'.
```
Comment (nico-kn) [inline sap_business_document_processing/document_information_extraction_client/dox_api_client.py:184]: Should we extend the docstring to hint to the constants that can be used to pass in this information?

File: sap_business_document_processing/document_information_extraction_client/constants.py [hunk 2]
```diff
@@ -36,7 +36,9 @@
 
 API_HEADER_ACCEPT = 'accept'
 
+CONTENT_TYPE_PDF = 'application/pdf'
 CONTENT_TYPE_PNG = 'image/png'
+CONTENT_TYPE_UNKNOWN = 'unknown'
```
Comment (nico-kn) [inline sap_business_document_processing/document_information_extraction_client/constants.py:41]: I think we should extend this list of constants to all of the supported content types. I.e. add jpeg, tiff and excel content types? https://help.sap.com/viewer/5fa7265b9ff64d73bac7cec61ee55ae6/SHIP/en-US/9e6a56d37ad1463fb3b85eed5d962a3c.html

File: sap_business_document_processing/document_information_extraction_client/dox_api_client.py [hunk 3]
```diff
@@ -196,24 +199,26 @@ def extract_information_from_document(self, document_path: str, client_id, docum
...
                                                        return_null_values: bool = False) -> dict:
         """
         Extracts the information from a document. The function will run until a processing result can be returned or
         a timeout is reached.
         :param document_path: The path to the document
         :param options: The options for processing the document as dictionary. It has to include at least a valid client
         ID and document type
-        :param mime_type: Content type of the uploaded file. If None is given, the content type is fetched automatically.
+        :param mime_type: Content type of the uploaded file. If 'unknown' is given, the content type is fetched
+        automatically. Default is 'application/pdf'.
```
Comment (nico-kn) [inline sap_business_document_processing/document_information_extraction_client/dox_api_client.py:212]: Same applies to this docstring

File: sap_business_document_processing/document_information_extraction_client/dox_api_client.py [hunk 4]
```diff
@@ -224,8 +229,10 @@ def extract_information_from_documents(self, document_paths: List[str], client_i
         :param document_paths: A list of paths to the documents
         :param client_id: The client ID for which the documents should be uploaded
         :param document_type: The type of the document being uploaded. For available document types see documentation
-        :param mime_type: Content type of the uploaded files. Can be a list of types for each document or a string to
-        use the same type for all documents. If None is given, the content type is fetched automatically.
+        :param mime_type: Content type that is used for all uploaded files. If 'unknown' is given, the content type is
+        fetched automatically. Default is 'application/pdf'.
+        :param mime_type_list: A list of content types for each file to be uploaded. Has to have the same length as
+        'document_paths'. If this parameter is given, 'mime_type' will be ignored.
```
Comment (nico-kn) [inline sap_business_document_processing/document_information_extraction_client/dox_api_client.py:235]: And to this. But the addition looks good :)

**Review by nico-kn — CHANGES_REQUESTED** [review]
Did you check if any additional wikis, documentation or demos would need to be updated?

File: sap_business_document_processing/document_information_extraction_client/constants.py (hunk 2 above)
Comment (tillgeissler) [reply sap_business_document_processing/document_information_extraction_client/constants.py:41]: Originally, we didn't want to add any constants that aren't used within the library, but I think we could add them

**Review by nico-kn — APPROVED** [review]
👍

**Pull request merged**